Shared/Supporting Files/Views/SamplesSearchView.swift

```diff
@@ -0,0 +1,138 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+
+struct SamplesSearchView: View {
+    /// The search query to highlight.
+    private let query: String
+    
+    /// The search result to display in the various sections.
+    private let searchResult: SearchResult
+    
+    var body: some View {
+        List {
+            if !searchResult.nameMatches.isEmpty {
+                Section("Name Results") {
+                    ForEach(searchResult.nameMatches, id: \.name) { sample in
+                        NavigationLink {
+                            SampleDetailView(sample: sample)
+                                .id(sample.name)
+                        } label: {
+                            SampleRow(sample: sample, query: query)
+                        }
+                    }
+                }
+            }
+            if !searchResult.descriptionMatches.isEmpty {
+                Section("Description Results") {
+                    ForEach(searchResult.descriptionMatches, id: \.name) { sample in
+                        NavigationLink {
+                            SampleDetailView(sample: sample)
+                                .id(sample.name)
+                        } label: {
+                            SampleRow(sample: sample, query: query)
+                        }
+                    }
+                }
+            }
+            if !searchResult.tagMatches.isEmpty {
+                Section("Tags Results") {
+                    ForEach(searchResult.tagMatches, id: \.name) { sample in
+                        NavigationLink {
+                            SampleDetailView(sample: sample)
+                                .id(sample.name)
+                        } label: {
+                            SampleRow(sample: sample, query: query)
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
```suggestion
        }
        .listStyle(.insetGrouped)
```

This would remove the disclosure indicators on iOS, unless we want users to be able to collapse the categories.

<img width="50%" src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/16397058/e32293ea-3ff3-48d8-a86e-7c296e5fad88">

Reply:
From this [blog post](https://sarunw.com/posts/swiftui-list-style/#sidebar) it seems to me that the most semantically suitable style here should be iOS: `insetGrouped` and iPadOS/macOS: `sidebar`. Apple didn't do that though. So I don't think it is a big deal.

I actually like to have the disclosure button, and by using the default "automatic" list style, maybe it is more future proof. 🙂